.github/workflows/litmus_smoke.yaml

```diff
@@ -0,0 +1,98 @@
+name: Litmus Smoke Test
+
+on:
+  push:
+    branches: [main]
+    paths-ignore: ['**.md', '**.erb', 'schemas/*']
+  pull_request:
+    type: [opened, reopened, edited]
+    paths-ignore: ['**.md', '**.erb', 'schemas/*']
+
+jobs:
+  ubuntu:
+    name: Ubuntu
+    runs-on: ubuntu-latest
+    env:
+      PLATFORMS: deb_puppet6
+    steps:
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5.x'
+      - name: Checkout MotD module
+        uses: actions/checkout@v2
+        with:
+          # Default fetch-depth is 1
+          repository: puppetlabs/puppetlabs-motd
+      - name: Modify Gemfile
+        uses: DamianReeves/write-file-action@v1.0
+        with:
+          path: Gemfile
+          # This action appends by default
+          contents: "gem 'bolt', require: false, git: 'https://github.com/${{ github.repository }}', branch: '${{ github.ref }}'"
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle config with development
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('lib/bolt/version.rb') }}-${{ hashFiles('Gemfile') }}-${{ hashFiles('bolt.gemspec') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Run smoke test
+        run: |
+          bundle exec rake spec_prep
+          bundle exec rake litmus:provision_list[default]
+          bundle exec rake litmus:install_agent
+          bundle exec rake litmus:install_module
+          bundle exec rake litmus:install_modules_from_directory[./spec/fixtures/modules]
+          bundle exec rake litmus:add_feature[test_123]
+          bundle exec rake litmus:acceptance:parallel
+          bundle exec rake litmus:tear_down
```

Comment:
It appears that the only difference between the two platforms is several environment variables and the litmus rake commands. Could we use an OS `matrix strategy`, set the environment variable declaration on the `run statements`, and an `if statement` for platform like in https://github.com/puppetlabs/puppet-vscode/blob/main/.github/workflows/vscode-ci.yml#L35 to condense this?